File: frontend_v2/types/shop.ts
export type ShopStatus = 'ACTIVE' | 'PENDING' | 'SUSPENDED' | 'REJECTED' | 'EXPIRED' | 'INACTIVE' | string;

export interface ShopBanner {
  id?: number;
  imageUrl: string;
  title?: string;
  subtitle?: string;
  buttonText?: string;
  buttonUrl?: string;
  displayOrder: number;
  isActive: boolean;
}

export interface ShopBusinessHours {
  id?: number;
  dayOfWeek: string; // MONDAY..SUNDAY
  isOpen: boolean;
  openTime: string; // "09:00"
  closeTime: string; // "21:00"
}

export interface ShopDeliveryConfig {
  id?: number;
  deliveryChargeType: 'FREE' | 'FIXED' | string;
  fixedChargeAmount: number;
  minOrderForFreeDelivery?: number | null;
  deliveryNotes?: string | null;
}

export interface ShopStorefrontSettings {
  id?: number;
  heroBannerEnabled: boolean;
  topRatedEnabled: boolean;
  reviewsEnabled: boolean;
  bakeryInfoEnabled: boolean;
  categoriesEnabled: boolean;
  filtersEnabled: boolean;
  ratingsEnabled: boolean;
  aboutStoryEnabled: boolean;
  aboutImageEnabled: boolean;
  fulfillmentEnabled: boolean;
  leadTimeDays: number;
  leadTimeMessage: string;
  customCakesEnabled: boolean;
  whatsappEnabled: boolean;
  phoneEnabled: boolean;
  emailEnabled: boolean;
  addressEnabled: boolean;
  mapEnabled: boolean;
  businessHoursEnabled: boolean;
}

export interface ShopCustomFormField {
  id?: number;
  fieldKey: string;
  fieldLabel: string;
  fieldType: 'TEXT' | 'TEXTAREA' | 'NUMBER' | 'DROPDOWN' | 'RADIO' | 'CHECKBOX' | 'DATE' | 'TIME' | 'IMAGE' | string;
  isRequired: boolean;
  isEnabled: boolean;
  optionsJson?: string | null;
  displayOrder: number;
}

export interface Shop {
  id: number;
  businessName: string;
  businessType?: string;
  businessCategory?: string;
  businessDescription?: string;
  description?: string;
  businessPhone?: string;
  businessEmail?: string;
  phone?: string;
  email?: string;
  address?: string;
  addressLine1: string;
  addressLine2?: string;
  area?: string;
  city: string;
  district?: string;
  state: string;
  pincode: string;
  distanceKm?: number;
  latitude?: number;
  longitude?: number;
  status: ShopStatus;
  verificationStatus?: 'VERIFIED' | 'PROCESSING' | 'UNVERIFIED' | string;
  fssaiRegistration?: string;
  yearsInBusiness?: number;
  rating?: number;
  reviewCount?: number;
  imageUrl?: string;
  bannerUrl?: string;
  coverImageUrl?: string;
  logoUrl?: string;
  isPureVeg?: boolean;
  deliveryAvailable?: boolean;
  deliveryTimeMinutes?: number;
  minOrderValue?: number;
  featuredCategory?: string;
  createdAt?: string;
  updatedAt?: string;

  // About & Social
  aboutStory?: string;
  aboutImageUrl?: string;
  showAboutImage?: boolean;
  whatsappNumber?: string;
  mapLocationUrl?: string;

  // Review Summary
  averageRating?: number;
  totalReviews?: number;

  // Storefront Configuration & Components
  banners?: ShopBanner[];
  businessHours?: ShopBusinessHours[];
  deliveryConfig?: ShopDeliveryConfig;
  storefrontSettings?: ShopStorefrontSettings;
  customCakeFormFields?: ShopCustomFormField[];
}

export interface ShopSearchFilters {
  state?: string;
  district?: string;
  city?: string;
  area?: string;
  pincode?: string;
  country?: string;
  businessType?: string;
  search?: string;
  location?: string;
  latitude?: number;
  longitude?: number;
  radiusKm?: number;
  sortBy?: string;
  page?: number;
  size?: number;
}

export interface PopularCity {
  cityName: string;
  stateName?: string;
  activeBakeryCount: number;
}

File: backend/src/main/java/com/cakeplatform/api/modules/shop/Shop.java
package com.cakeplatform.api.modules.shop;

import com.cakeplatform.api.modules.user.User;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "shops")
@Data
public class Shop {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @Column(name = "business_name", nullable = false)
    private String businessName;

    private String description;
    private String phone;
    private String email;
    private String address;
    private String city;
    private String state;
    private String pincode;

    @Column(name = "business_category")
    private String businessCategory;

    @Enumerated(EnumType.STRING)
    @Column(name = "business_type")
    private BusinessType businessType;

    @Column(name = "years_in_business")
    private Integer yearsInBusiness;

    @Column(name = "fssai_registration")
    private String fssaiRegistration;

    @Column(name = "address_line_1")
    private String addressLine1;

    @Column(name = "address_line_2")
    private String addressLine2;

    private String area;
    private String district;

    @Column(name = "latitude")
    private Double latitude;

    @Column(name = "longitude")
    private Double longitude;

    @Column(name = "logo_url")
    private String logoUrl;

    @Column(name = "cover_image_url")
    private String coverImageUrl;

    @Column(name = "about_story", columnDefinition = "TEXT")
    private String aboutStory;

    @Column(name = "about_image_url", length = 1000)
    private String aboutImageUrl;

    @Column(name = "show_about_image", nullable = false)
    private Boolean showAboutImage = true;

    @Column(name = "whatsapp_number", length = 50)
    private String whatsappNumber;

    @Column(name = "map_location_url", length = 1000)
    private String mapLocationUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ShopStatus status = ShopStatus.PENDING;

    @Enumerated(EnumType.STRING)
    @Column(name = "verification_status")
    private VerificationStatus verificationStatus = VerificationStatus.PROCESSING;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "shop", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ShopDeliverySlot> deliverySlots = new ArrayList<>();

    @OneToOne(mappedBy = "shop", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private ShopDeliveryConfig deliveryConfig;

    @OneToOne(mappedBy = "shop", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private ShopStorefrontSettings storefrontSettings;

    @OneToMany(mappedBy = "shop", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("displayOrder ASC, createdAt ASC")
    private List<ShopBanner> banners = new ArrayList<>();

    @OneToMany(mappedBy = "shop", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ShopBusinessHours> businessHours = new ArrayList<>();

    @OneToMany(mappedBy = "shop", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("displayOrder ASC")
    private List<ShopCustomFormField> customFormFields = new ArrayList<>();
}
